2020/day1/solution.py

```diff
@@ -1,15 +1,14 @@
 file = open('input.txt')
```

Comment:
```suggestion
with open('input.txt') as f:
```
(You'll need to indent the next line by 4 as well, GH won't let me comment on it. This looks to apply to most solution files.)
Best practice is to use [context managers](https://docs.python.org/3/library/stdtypes.html#context-manager-types) whenever you are managing resources to ensure cleanup happens promptly so you don't e.g. run out of file descriptors.

Overkill for this code, but https://docs.python.org/3/library/pathlib.html provides an alternative:
```suggestion
from pathlib import Path
contents = Path('input.txt').read_text()
```
It's more useful when the filename is dynamic where pathlib makes it easier to compute. Also handy in tests to save some lines - I use this with pytest's [`tmp_path` fixture](https://docs.pytest.org/en/latest/tmpdir.html) a lot.

Reply:
ooh,  good old raii :)  I didn't know python had that